### PR TITLE
Add ByteBuffer-based put-method to SstFileWriter

### PR DESCRIPTION
**Pull request by zawlazaw**
This PR adds another variant of the put-method to SstFileWriter, namely with the signature `put(ByteBuffer, ByteBuffer)`.
This allows for the most efficient data transfer from Java to rocksdb.
See issue #2668 for a detailed motivation.

### CODE REVIEW

**Comment by gfosco** [comment]
Closing this via automation due to lack of activity.  If you'd like this to be re-opened, please just comment here and we'll open it back up.

**Comment by adamretter** [comment]
I think we should re-review this after we get https://github.com/facebook/rocksdb/pull/2283 into the code base. As I think there is some overlap

**Comment by unoexperto** [comment]
@adamretter Looks like #2283 is merged :)

**Comment by adamretter** [comment]
@zawlazaw could we get this rebased?

**Comment by zawlazaw** [comment]
I rebased locally, and the conflicts showed that my PR is obsolte now:

My Java code introduces three methods:
1. `public void put(final ByteBuffer key, final ByteBuffer value)` to `SstFileWriter.java`
2. `private native void put(...)` to `SstFileWriter.java`
3. `Java_org_rocksdb_SstFileWriter_put__JLjava_nio_ByteBuffer_2IILjava_nio_ByteBuffer_2II(...)` to `sst_file_writerjni.cc`

Which add essentially the same functionality and API as now already can be found on master:
1. https://github.com/facebook/rocksdb/blob/4cbc19d2a1da1a44775f27c2d167a0d8da0fd9f5/java/src/main/java/org/rocksdb/SstFileWriter.java#L132
2. https://github.com/facebook/rocksdb/blob/4cbc19d2a1da1a44775f27c2d167a0d8da0fd9f5/java/src/main/java/org/rocksdb/SstFileWriter.java#L270
3. https://github.com/facebook/rocksdb/blob/4cbc19d2a1da1a44775f27c2d167a0d8da0fd9f5/java/rocksjni/sst_file_writerjni.cc#L152

So I guess my PR can be closed without merging.

File: java/src/main/java/org/rocksdb/SstFileWriter.java [hunk 1]
```diff
@@ -126,10 +128,24 @@ public void put(final DirectSlice key, final DirectSlice value)
...
+  /**
+   * Add a Put key with value to currently opened file.
+   *
+   * @param key the specified key to be inserted (reading from position to limit).
+   * @param value the value associated with the specified key (reading from position to limit).
+   *
+   * @throws RocksDBException thrown if error happens in underlying
+   *    native library.
+   */
+  public void put(final ByteBuffer key, final ByteBuffer value)
```
Comment (javeme) [inline java/src/main/java/org/rocksdb/SstFileWriter.java:145]: seems it has been implemented by https://github.com/facebook/rocksdb/commit/e412a426d6ce019fb12f094f64e7f105d227201d . https://github.com/facebook/rocksdb/blob/2e09a54c4fb82e88bcaa3e7cfa8ccbbbbf3635d5/java/src/main/java/org/rocksdb/SstFileWriter.java#L134

**Review by javeme — COMMENTED** [review]
seems this pr could be closed since it has been implemented by #2283

**Comment by mrambacher** [comment]
Closing, since this seems resolved by #2283

**Pull request closed without merging**